app/src/main/java/mozilla/lockbox/support/TimingSupport.kt

```diff
@@ -17,33 +17,50 @@ import mozilla.lockbox.store.ContextStore
 import mozilla.lockbox.store.SettingStore
 
 @ExperimentalCoroutinesApi
-open class AutoLockSupport(
+open class TimingSupport(
     val settingStore: SettingStore = SettingStore.shared
 ) : ContextStore {
 
     companion object {
-        val shared by lazy { AutoLockSupport() }
+        val shared by lazy { TimingSupport() }
     }
 
     private val compositeDisposable = CompositeDisposable()
     private lateinit var preferences: SharedPreferences
 
-    var lockingSupport: LockingSupport = SystemLockingSupport()
+    var systemTimingSupport: SystemTimingSupport = SystemSystemTimingSupport()
 
     open val shouldLock: Boolean
         get() = lockCurrentlyRequired()
 
+    open val shouldSync: Boolean
+        get() = syncCurrentlyRequired()
+
     override fun injectContext(context: Context) {
         preferences = PreferenceManager.getDefaultSharedPreferences(context)
     }
 
+    private fun syncCurrentlyRequired(): Boolean {
+        val syncTimerDate = preferences.getLong(Constant.Key.syncTimerDate, Long.MAX_VALUE)
+        val currentSystemTime = systemTimingSupport.systemTimeElapsed
+
+        // this will be triggered when the system time is very low due to device restarts
+        val currentTimeTooEarly = (syncTimerDate - currentSystemTime) > Constant.Common.twentyFourHours
+
+        return syncTimerDate < currentSystemTime || currentTimeTooEarly
```

Comment:
Not sure I understand the name and comment `currentTimeTooEarly`.

```kotlin
syncTimerDate > currentSystemTime + twentyFourHours
```

Oh, so if `syncTimerDate` (which should never be less than twenty four hours after the current time) is 24h after the current time, then we've had something weird happen, e.g. a reboot.

So, suggest: renaming `currentTimeTooEarly` to `followingDeviceReboot`.

```kotlin
val followingDeviceReboot = syncTimerDate > currentSystemTime + twentyFourHours
```

Reply:
Incidentally, I'd put 24 hours as a property.